js/ui/windowManager.js

```diff
@@ -717,6 +784,126 @@ const WindowManager = new Lang.Class({
 
         global.display.connect('grab-op-begin', Lang.bind(this, this._windowGrabbed));
         global.display.connect('grab-op-end', Lang.bind(this, this._windowUngrabbed));
+
+        this._sylvesterListener = new SylvesterService(Gio.DBus.session,
+                                                       'com.endlessm.Sylvester.Service',
+                                                       '/com/endlessm/Sylvester/Service');
+        this._sylvesterListener.connectSignal('RotateBetweenPidWindows',
+                                              Lang.bind(this, this._handleRotateBetweenPidWindows));
+        this._pendingRotateAnimations = [];
+        this._rotateOutActors = [];
+        this._rotateInActors = [];
+        this._rotateInTimeouts = [];
+        log("Connected to Sylvester");
+    },
+
+    _handleRotateBetweenPidWindows: function(proxy, sender, [src, dst]) {
+        const srcActorInfo = pidToActorInfo(src)
+        this._pendingRotateAnimations.push({
+            src: srcActorInfo,
+            dst: pidToActorInfo(dst)
+        });
+        this._updateReadyRotateAnimationsWith(srcActorInfo.actor);
+    },
+    
+    _updateReadyRotateAnimationsWith: function (window) {
+        /* A new window was added. Get its pid and look for any
+         * unsatisfied entries in _pendingRotateAnimations */
+        const pid = window ? window.get_meta_window().get_pid() : null;
+        const lastPendingRotateAnimationsLength = this._pendingRotateAnimations.length;
+        this._pendingRotateAnimations = this._pendingRotateAnimations.filter(Lang.bind(this, function(animationSpec) {
+            let unsatisfiedPids = 0;
+            Object.keys(animationSpec).forEach(function(key) {
+                if (animationSpec[key].window == null) {
+                    if (animationSpec[key].pid == pid) {
+                        animationSpec[key].window = window;
+                    } else {
+                        unsatisfiedPids++;
+                    }
+                }
+            });
+            
+            if (unsatisfiedPids == 0) {
+                /* Delaying the animation like this is not ideal, but it is necessary
+                 * to allow some time for the window to paint itself for the first time.
+                 *
+                 * Mutter doesn't seem to have any kind of notification for when we first
```

Comment:
Mutter has a `first-frame` signal that it emits to track this specific case. Not sure if it's helpful here.